client/src/main/java/tds/assessment/AssessmentInfo.java

```diff
@@ -0,0 +1,186 @@
+package tds.assessment;
+
+import java.util.List;
+
+/**
+ * An object containing assessment metadata
+ */
+public class AssessmentInfo {
+    private String key;
+    private String id;
+    private String subject;
+    private String label;
+    private int maxAttempts;
+    private List<String> grades;
+    private List<String> languages;
+    private List<AssessmentWindow> assessmentWindows;
+
+    /**
+     * Empty constructor for frameworks
+     */
+    private AssessmentInfo() {}
+
+    public AssessmentInfo(Builder builder) {
+        this.maxAttempts = builder.maxAttempts;
+        this.assessmentWindows = builder.assessmentWindows;
+        this.id = builder.id;
+        this.label = builder.label;
+        this.subject = builder.subject;
+        this.grades = builder.grades;
+        this.languages = builder.languages;
+        this.key = builder.key;
+    }
+
+    public static final class Builder {
+        private String key;
+        private String id;
+        private String subject;
+        private String label;
+        private int maxAttempts;
+        private List<String> grades;
+        private List<String> languages;
+        private List<AssessmentWindow> assessmentWindows;
+
+        public Builder withKey(String key) {
+            this.key = key;
+            return this;
+        }
+
+        public Builder withId(String id) {
+            this.id = id;
+            return this;
+        }
+
+        public Builder withSubject(String subject) {
+            this.subject = subject;
+            return this;
+        }
+
+        public Builder withLabel(String label) {
+            this.label = label;
+            return this;
+        }
+
+        public Builder withMaxAttempts(int maxAttempts) {
+            this.maxAttempts = maxAttempts;
+            return this;
+        }
+
+        public Builder withGrades(List<String> grades) {
+            this.grades = grades;
+            return this;
+        }
+
+        public Builder withLanguages(List<String> languages) {
+            this.languages = languages;
+            return this;
+        }
+
+        public Builder withAssessmentWindows(List<AssessmentWindow> assessmentWindows) {
+            this.assessmentWindows = assessmentWindows;
+            return this;
+        }
+
+        public Builder fromAssessmentInfo(AssessmentInfo assessmentInfo) {
```

Comment:
Can we make this static and return a new builder?  `public static Builder fromAssessmentInfo(info)`